File: spec/fizzbuzz_spec.rb
require "./jasmine/lib/fizzbuzz"

describe 'fizzbuzz' do
  it "returns 'fizz' when passed 3" do
    expect(fizzbuzz(3)).to eq 'fizz'
  end

  it "returns 'buzz' when passed 5" do
    expect(fizzbuzz(5)).to eq "buzz"
  end

  it "returns 'fizzbuzz' when passed 15" do
    expect(fizzbuzz(15)).to eq "fizzbuzz"
  end

  it "returns 1 when passed 1" do
    expect(fizzbuzz(1)).to eq 1
  end

  it "returns 'fizz' when given multiple of 3" do
    expect(fizzbuzz(6)).to eq "fizz"
    expect(fizzbuzz(9)).to eq "fizz"
  end

  it "returns 'buzz' when given multiple of 5" do
    expect(fizzbuzz(10)).to eq "buzz"
    expect(fizzbuzz(20)).to eq "buzz"
    expect(fizzbuzz(25)).to eq "buzz"
  end

  it "returns 'fizzbuzz' when given multiples of 3 and 5" do
    expect(fizzbuzz(15)).to eq "fizzbuzz"
    expect(fizzbuzz(30)).to eq "fizzbuzz"
    expect(fizzbuzz(45)).to eq "fizzbuzz"
    expect(fizzbuzz(600)).to eq "fizzbuzz"
    expect(fizzbuzz(3000)).to eq "fizzbuzz"
  end

  it "returns number when given number not multiple of 3 or 5" do
    expect(fizzbuzz(7)).to eq 7
    expect(fizzbuzz(28)).to eq 28
    expect(fizzbuzz(613544)).to eq 613544
    expect(fizzbuzz(0)).to eq 0
  end

  it "returns 'fizz' when given multiple of -3" do
    expect(fizzbuzz(-3)).to eq 'fizz'
    expect(fizzbuzz(-6)).to eq 'fizz'
    expect(fizzbuzz(-9)).to eq 'fizz'
  end

  it "returns 'buzz' when given multiple of -5" do
    expect(fizzbuzz(-10)).to eq 'buzz'
    expect(fizzbuzz(-20)).to eq 'buzz'
    expect(fizzbuzz(-25)).to eq 'buzz'
  end

  it "returns 'fizzbuzz' when given negative multiples of 3 and 5" do
    expect(fizzbuzz(-15)).to eq 'fizzbuzz'
    expect(fizzbuzz(-30)).to eq 'fizzbuzz'
    expect(fizzbuzz(-150)).to eq 'fizzbuzz'
  end

  it "returns nil when given nil" do
    expect(fizzbuzz(nil)).to eq nil
  end

  it "returns number when given 'number' as a string" do
    expect(fizzbuzz("1")).to eq 1
    expect(fizzbuzz("0")).to eq 0
  end

  it "returns 'fizz' when given multiple of '3' as a string" do
    expect(fizzbuzz("3")).to eq "fizz"
    expect(fizzbuzz("6")).to eq "fizz"
    expect(fizzbuzz("72")).to eq "fizz"
  end

  it "returns nil when given input that cannot be coerced into an integer" do
    expect(fizzbuzz("gibberish")).to eq nil
    expect(fizzbuzz(Class.new)).to eq nil
    expect(fizzbuzz([])).to eq nil
  end

  it "returns 'fizz' when passed 3.0" do
    expect(fizzbuzz(3.0)).to eq "fizz"
  end

  it "returns float when passed float" do
    expect(fizzbuzz(1.0)).to eq 1.0
    expect(fizzbuzz(1.5)).to eq 1.5
  end
end

File: jasmine/spec/fizzBuzzSpec.js
describe("FizzBuzz", function() {
  it("should return 'fizz' when passed multiples of 3", function(){
    expect(FizzBuzz(3)).toEqual('fizz');
    expect(FizzBuzz(6)).toEqual("fizz");
    expect(FizzBuzz(9)).toEqual("fizz");
    expect(FizzBuzz(303)).toEqual('fizz');
  });

  it("should return 'buzz' when passed multiples of 5", function(){
    expect(FizzBuzz(5)).toEqual("buzz");
    expect(FizzBuzz(10)).toEqual("buzz");
    expect(FizzBuzz(50)).toEqual("buzz");
    expect(FizzBuzz(500)).toEqual("buzz");
    expect(FizzBuzz(6846155)).toEqual('buzz');
  });

  it("returns 'fizzbuzz' when passed multiples of 15", function(){
    expect(FizzBuzz(15)).toEqual("fizzbuzz");
    expect(FizzBuzz(30)).toEqual("fizzbuzz");
    expect(FizzBuzz(150)).toEqual('fizzbuzz');
  });

  it("returns the number when not divisible by 3 or 5", function (){
    expect(FizzBuzz(1)).toEqual(1);
    expect(FizzBuzz(2)).toEqual(2);
    expect(FizzBuzz(7)).toEqual(7);
    expect(FizzBuzz(92)).toEqual(92);
  });
});
